backend/routers/social_media_marketing_router.py

```
@@ -224,6 +224,98 @@ async def upload_swamiji_image(
+        )
+        
+        # Update the database
+        await conn.execute(
+            "INSERT INTO platform_settings (key, value) VALUES ('swamiji_avatar_url', $1) ON CONFLICT (key) DO UPDATE SET value = EXCLUDED.value, updated_at = NOW()",
+            json.dumps(public_url)
+        )
+
+        return StandardResponse(success=True, message="Master avatar has been set successfully.", data={"new_avatar_url": public_url})
+    except Exception as e:
+        logger.error(f"Failed to set master avatar: {e}", exc_info=True)
+        raise HTTPException(status_code=500, detail="Failed to set master avatar.")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)